Docs: SAML role and team sync (#23986)
grafana/grafana: docs/sources/enterprise/saml.md

To be able to use Team Sync feature, set `assertion_attribute_groups` option to the attribute name where you store user groups. Then Grafana will use attribute values extracted from SAML assertion to add user into the groups with the same name configured on the External group sync tab.

[Learn more about Team Sync]({{< relref "../enterprise/team-sync.md" >}})

### Configure role sync

> Only available in Grafana v7.0+

Role sync allows to map user roles from identity provider to Grafana. To enable role sync, set `assertion_attribute_role` option to the attribute name where the role information will be extracted from. After that, configure desired attribute values for each of Grafana roles: Editor, Admin and Super Admin. If obtained role doesn't match any of configured values, `Viewer` role will be assigned to the user. Example configuration:

```bash
[auth.saml]
assertion_attribute_role = role
role_values_editor = editor, developer
role_values_admin = admin, operator
role_values_grafana_admin = superadmin
```

Refer to [Organization roles]({{< relref "../permissions/organization_roles.md" >}}) for more information about roles and permissions in Grafana.

> **Important**: when role sync is configured, any changes of user roles and organization membership made in Grafana will be overwritten on next user login. Assign user organizations and roles in the IdP instead.

Reviewer: ```suggestion
**Important**: when role sync is configured, any changes of user roles and organization membership made manually in Grafana will be overwritten on next user login. Assign user organizations and roles in the IdP instead.
```

### Configure organization mapping

> Only available in Grafana v7.0+

Organization mapping allows you to assign users to particular organization in Grafana depending on configured `assertion_attribute_org` attribute value obtained from identity provider. Set `assertion_attribute_org` to the attribute name you store organization info in. Then configure `org_mapping` option to map organization from IdP to Grafana organization specified by id. For example, `org_mapping = Engineering:2, Sales:3` assigns users from `Engineering` organization to the Grafana organization with id `2` and users from `Sales` - to the org with id `3`. You can specify multiple organizations both for the IdP and Grafana:

> 

Reviewer: I think we should show setting `assertion_attribute_org` here too to clearly show that it's also needed.

Reviewer 2: @xlson , does this look good to you?

Reviewer: I think that looks great, a lot easier to follow now.

* `org_mapping = Engineering:2, Sales:2` to map users from `Engineering` and `Sales` to `2` in Grafana.
* `org_mapping = Engineering:2, Engineering:3` to assign `Engineering` to both `2` and `3` in Grafana.

### Configure allowed organizations

> Only available in Grafana v7.0+

> With `allowed_organizations` option you can specify a list of organizations user must have a membership of to be able to log in into Grafana.

Reviewer: ```suggestion
With the `allowed_organizations` option you can specify a list of organizations where the user must be a member of at least one of them to be able to log in to Grafana.
```

An example is `assertion_attribute_name = "givenName"` where Grafana looks within the assertion for an attribute with a friendly name or name of `givenName`. Both, the friendly name (e.g. `givenName`) or the name (e.g. `urn:oid:2.5.4.42`) can be used interchangeably as the value for the configuration option.

> ### Configure team sync

Reviewer 2: ```suggestion
### Configure Team Sync
```

Author: We don't have **Team Sync** in UI and at the other pages we use **Team sync** in sentence case.

> > Only available in Grafana v7.0+

Reviewer 2: This is misleading. Team Synch was available before 7.0. Please clarify what functionality this refers to.

Author: This refers to SAML only. Probably,
> Team sync support for SAML only available in Grafana v7.0+

Reviewer: SAML did not support Team Sync before 7.0 so this is accurate. Perhaps we can make that more clear somehow?

Reviewer 2: I like @alexanderzobnin 's suggestion. Let's go with that.

Reviewer 2: Looks like this still needs updating.

> To be able to use Team Sync feature, set `assertion_attribute_groups` option to the attribute name where you store user groups. Then Grafana will use attribute values extracted from SAML assertion to add user into the groups with the same name configured on the External group sync tab.

Reviewer 2: ```suggestion
To use SAML Team Sync, set `assertion_attribute_groups` to the attribute name where you store user groups. Then Grafana will use attribute values extracted from SAML assertion to add user into the groups with the same name configured on the External group sync tab.
```

Reviewer 2: Two things: 
* Link to the section in configuration.md that talks about this setting.
* Make sure the "External group sync tab" text capitalization matches the UI.

Author: Yes, **External group sync** is the exact name of the tab:

![Screenshot from 2020-04-29 08-47-08](https://user-images.githubusercontent.com/4932851/80564256-08c04580-89f6-11ea-80e9-08bc88d4c394.png)

Author: Regarding `configuration.md`: we don't have SAML and other enterprise-specific configs there. But at this page I added all SAML configuration options with a detailed description to the table (**Set up SAML authentication** section).

Reviewer: Good point, we might want to make a configuration.md in the Enterprise parts of the docs that collects all the Enterprise specific settings.

Reviewer 2: I don't have an Enterprise license, so if someone could do that, it would be great. Definitely a gap in the documentation that we should fill.

[Learn more about Team Sync]({{< relref "../enterprise/team-sync.md" >}})

> ### Configure role sync

Reviewer 2: Is "role sync" an actual name, present in the UI? If so, match UI capitalization.

Author: No, there's no feature in the UI, it's a kind of configuration option.

> Only available in Grafana v7.0+

> Role sync allows to map user roles from identity provider to Grafana. To enable role sync, set `assertion_attribute_role` option to the attribute name where the role information will be extracted from. After that, configure desired attribute values for each of Grafana roles: Editor, Admin and Super Admin. If obtained role doesn't match any of configured values, `Viewer` role will be assigned to the user. Example configuration:

Reviewer 2: ```suggestion
Role sync allows you to map user roles from an identity provider to Grafana. 

To enable role sync, set the `assertion_attribute_role` option to the attribute name where the role information will be extracted from. After that, configure desired attribute values for each Grafana role: Editor, Admin, and Super Admin. If the obtained role doesn't match any of configured values, then the `Viewer` role is assigned to the user. Example configuration:
```

> Role sync allows to map user roles from identity provider to Grafana. To enable role sync, set `assertion_attribute_role` option to the attribute name where the role information will be extracted from. After that, configure desired attribute values for each of Grafana roles: Editor, Admin and Super Admin. If obtained role doesn't match any of configured values, `Viewer` role will be assigned to the user. Example configuration:

Reviewer 2: Need links to the appropriate sections in the Permissions section (for roles) and Configuaration.

Also, what is "Super Admin"? I do not recall this mentioned anywhere in our current documentation.

Author: This is a Grafana Admin also called Super Admin. That mentioned at the [permissions overview](https://grafana.com/docs/grafana/latest/permissions/overview/#grafana-admin) page. Probably, we can change it to `Grafana Admin` if it's more clear.

```bash
[auth.saml]
assertion_attribute_role = role
role_values_editor = editor, developer
role_values_admin = admin, operator
role_values_grafana_admin = superadmin
```

> Refer to [Organization roles]({{< relref "../permissions/organization_roles.md" >}}) for more information about roles and permissions in Grafana.

Reviewer 2: I suggest moving this above the example.

> **Important**: when role sync is configured, any changes of user roles and organization membership made manually in Grafana will be overwritten on next user login. Assign user organizations and roles in the IdP instead.

Reviewer 2: ```suggestion
**Important**: When role sync is configured, any changes of user roles and organization membership made manually in Grafana will be overwritten on next user login. Assign user organizations and roles in the IdP instead.
```

Reviewer 2: Where is the IdP? What does this acronym stand for?

Author: At the top of the page we have
>SAML authentication integration allows your Grafana users to log in by using an external SAML 2.0 Identity Provider (IdP)

### Configure organization mapping

> Only available in Grafana v7.0+

> Organization mapping allows you to assign users to particular organization in Grafana depending on configured `assertion_attribute_org` attribute value obtained from identity provider. Set `assertion_attribute_org` to the attribute name you store organization info in. Then configure `org_mapping` option to map organization from IdP to Grafana organization specified by id. For example, use following config to assign users from `Engineering` organization to the Grafana organization with id `2` and users from `Sales` - to the org with id `3`, based on `Org` assertion attribute value:

Reviewer 2: ```suggestion
Organization mapping allows you to assign users to a particular organization in Grafana depending on the `assertion_attribute_org` attribute value obtained from the identity provider. Set the `assertion_attribute_org` to the attribute name you store organization info in, then configure `org_mapping` option to map organization from IdP to Grafana organization specified by id. 

For example, use the following configuration to assign users from `Engineering` organization to the Grafana organization with id `2` and users from `Sales` - to the org with id `3`, based on `Org` assertion attribute value:
```

An example is `assertion_attribute_name = "givenName"` where Grafana looks within the assertion for an attribute with a friendly name or name of `givenName`. Both, the friendly name (e.g. `givenName`) or the name (e.g. `urn:oid:2.5.4.42`) can be used interchangeably as the value for the configuration option.

### Configure team sync

> Only available in Grafana v7.0+

To use SAML Team sync, set `assertion_attribute_groups` to the attribute name where you store user groups. Then Grafana will use attribute values extracted from SAML assertion to add user into the groups with the same name configured on the External group sync tab.

[Learn more about Team Sync]({{< relref "../enterprise/team-sync.md" >}})

### Configure role sync

> Only available in Grafana v7.0+

Role sync allows you to map user roles from an identity provider to Grafana. To enable role sync, configure role attribute and possible values for [Editor]({{< relref "../permissions/organization_roles.md#editor-role" >}}), [Admin]({{< relref "../permissions/organization_roles.md#admin-role" >}}) and [Grafana Admin]({{< relref "../permissions/overview.md#grafana-admin" >}}) roles.

> 1. In configuration file, set `assertion_attribute_role` option to the attribute name where the role information will be extracted from.

Reviewer 2: ```suggestion
1. In the configuration file, set `assertion_attribute_role` option to the attribute name where the role information will be extracted from.
```

> 1. Set `role_values_editor` option to the values mapped to the `Editor` role.

Reviewer 2: For each of these steps, please edit it this way so that it is correct:

Set the ___ option to...

1. Set `role_values_admin` option to the values mapped to the organization `Admin` role.
1. Set `role_values_grafana_admin` option to the values mapped to the `Grafana Admin` role.

> 1. If user roles doesn't match any of configured values, `Viewer` role will be assigned.

Reviewer 2: ```suggestion
  If a user role doesn't match any of configured values, then the `Viewer` role will be assigned.
```

### Configure role sync

> Only available in Grafana v7.0+

Role sync allows you to map user roles from an identity provider to Grafana. To enable role sync, configure role attribute and possible values for [Editor]({{< relref "../permissions/organization_roles.md#editor-role" >}}), [Admin]({{< relref "../permissions/organization_roles.md#admin-role" >}}) and [Grafana Admin]({{< relref "../permissions/overview.md#grafana-admin" >}}) roles.

1. In the configuration file, set [`assertion_attribute_role`]({{< relref "./enterprise-configuration.md#assertion-attribute-role" >}}) option to the attribute name where the role information will be extracted from.
1. Set the [`role_values_editor`]({{< relref "./enterprise-configuration.md#role-values-editor" >}}) option to the values mapped to the `Editor` role.
1. Set the [`role_values_admin`]({{< relref "./enterprise-configuration.md#role-values-admin" >}}) option to the values mapped to the organization `Admin` role.
1. Set the [`role_values_grafana_admin`]({{< relref "./enterprise-configuration.md#role-values-grafana-admin" >}}) option to the values mapped to the `Grafana Admin` role.

> If a user role doesn't match any of configured values, then the `Viewer` role will be assigned.

Reviewer: ```suggestion
If a user role doesn't match any of the configured values, then the `Viewer` role will be assigned.
```

Refer to [Organization roles]({{< relref "../permissions/organization_roles.md" >}}) for more information about roles and permissions in Grafana.

Example configuration:

```bash
[auth.saml]
assertion_attribute_role = role
role_values_editor = editor, developer
role_values_admin = admin, operator
role_values_grafana_admin = superadmin
```

**Important**: When role sync is configured, any changes of user roles and organization membership made manually in Grafana will be overwritten on next user login. Assign user organizations and roles in the IdP instead.

### Configure organization mapping

> Only available in Grafana v7.0+

> Organization mapping allows you to assign users to particular organization in Grafana depending on attribute value obtained from identity provider. 

Reviewer: ```suggestion
Organization mapping allows you to assign users to particular organization in Grafana depending on an attribute value obtained from the identity provider. 
```

> 1. In configuration file, set [`assertion_attribute_org`]({{< relref "./enterprise-configuration.md#assertion-attribute-org" >}}) to the attribute name you store organization info in.

Reviewer: ```suggestion
1. In the configuration file, set [`assertion_attribute_org`]({{< relref "./enterprise-configuration.md#assertion-attribute-org" >}}) to the attribute name you store the organization info in.
```

> 1. Set [`org_mapping`]({{< relref "./enterprise-configuration.md#org-mapping" >}}) option to the comma-separated list of `Organization:OrgId` pairs to map organization from IdP to Grafana organization specified by id.

Reviewer: ```suggestion
1. Set the [`org_mapping`]({{< relref "./enterprise-configuration.md#org-mapping" >}}) option to the comma-separated list of `Organization:OrgId` pairs to map organization from IdP to Grafana organization specified by id.
```

> For example, use following config to assign users from `Engineering` organization to the Grafana organization with id `2` and users from `Sales` - to the org with id `3`, based on `Org` assertion attribute value:

Reviewer: ```suggestion
For example, use the following config to assign users from the `Engineering` organization to the Grafana organization with id `2` and users from `Sales` - to the org with id `3`, based on the `Org` assertion attribute value:
```
